### ISSUE TITLE
value of "filedpath" in environment variables is empty in kubernetes dashboard

### ISSUE THREAD

**Issue by irniaxsep** [kubernetes/dashboard#5443]
The pod template:

apiVersion: apps/v1
kind: DaemonSet
metadata:
  name: busybox
  namespace: kube-system
  labels:
    k8s-app: busybox
spec:
  selector:
    matchLabels:
      name: busybox
  template:
    metadata:
      labels:
        name: busybox
    spec:
      containers: 
      - name: busybox
        image: k8s.gcr.io/busybox
        env:
          **- name: MY_NODE_NAME
            valueFrom:
              fieldRef:
                fieldPath: status.hostIP**

However, on kubernetes dashboard the value is empty:
![image](https://user-images.githubusercontent.com/9164629/90491601-b6f12600-e172-11ea-8073-43d48572aecb.png)

Will the host IP displayed on the dashboard.

**Comment by floreks** [comment]
If it's empty, then it means... well, that it is empty.

Try this test pod and see that `fieldPath` is displayed properly.

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: env-variables
spec:
  containers:
    - name: test-container
      image: k8s.gcr.io/busybox
      command: [ "sh", "-c"]
      args:
      - while true; do
          echo -en '\n';
          printenv MY_POD_NAME;
          printenv MY_MEM_REQUEST MY_MEM_LIMIT;
          printenv MY_CONST_VALUE;
          printenv MY_SECRET_VALUE;
          sleep 10;
        done;
      resources:
        requests:
          memory: "32Mi"
          cpu: "125m"
        limits:
          memory: "64Mi"
          cpu: "250m"
      env:
        - name: MY_POD_NAME
          valueFrom:
            fieldRef:
              fieldPath: metadata.name
        - name: MY_MEM_REQUEST
          valueFrom:
            resourceFieldRef:
              containerName: test-container
              resource: requests.memory
        - name: MY_MEM_LIMIT
          valueFrom:
            resourceFieldRef:
              containerName: test-container
              resource: limits.memory
```

**Comment by irniaxsep** [comment]
Thanks @floreks . I'll give it a try.  I did check the pod yaml file after deployed, the host IP is exists according to the output :

irina@cn-yuyu2-225:~$ kubectl get pod busybox-vtbvw -n kube-system -o yaml
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2020-08-18T03:19:04Z"
  generateName: busybox-
  labels:
    controller-revision-hash: 56d7844b79
    name: busybox
    pod-template-generation: "1"
  name: busybox-vtbvw
  namespace: kube-system
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: DaemonSet
    name: busybox
    uid: e1c0c1b1-efac-4289-8a7a-185e503c2e21
  resourceVersion: "383875"
  selfLink: /api/v1/namespaces/kube-system/pods/busybox-vtbvw
  uid: 66404efb-f033-47de-9be3-5c2ee474a025
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchFields:
          - key: metadata.name
            operator: In
            values:
            - aks-agentpool-18453427-vmss000001
  containers:
  - env:
    - name: MY_NODE_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: status.hostIP
    - name: KUBERNETES_PORT_443_TCP_ADDR
      value: aks0814-dns-00b2f2d6.hcp.eastus.azmk8s.io
    - name: KUBERNETES_PORT
      value: tcp://aks0814-dns-00b2f2d6.hcp.eastus.azmk8s.io:443
    - name: KUBERNETES_PORT_443_TCP
      value: tcp://aks0814-dns-00b2f2d6.hcp.eastus.azmk8s.io:443
    - name: KUBERNETES_SERVICE_HOST
      value: aks0814-dns-00b2f2d6.hcp.eastus.azmk8s.io
    image: k8s.gcr.io/busybox
    imagePullPolicy: Always
    name: busybox
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: default-token-qmljd
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  nodeName: aks-agentpool-18453427-vmss000001
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: default
  serviceAccountName: default
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/pid-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/unschedulable
    operator: Exists
  volumes:
  - name: default-token-qmljd
    secret:
      defaultMode: 420
      secretName: default-token-qmljd
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2020-08-18T03:19:04Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2020-08-18T03:19:04Z"
    message: 'containers with unready status: [busybox]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2020-08-18T03:19:04Z"
    message: 'containers with unready status: [busybox]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2020-08-18T03:19:04Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: docker://73e60d2fdb1b322efbd1fdada2d02b7b5e8f2a8b90a9e1a0dfd51516e772d637
    image: k8s.gcr.io/busybox:latest
    imageID: docker-pullable://k8s.gcr.io/busybox@sha256:d8d3bc2c183ed2f9f10e7258f84971202325ee6011ba137112e01e30f206de67
    lastState:
      terminated:
        containerID: docker://f55f957f3df5cd75380b6735b3719a39a1fe5e77bd57c8dcc0ab28ca90dc4dc9
        exitCode: 0
        finishedAt: "2020-08-18T03:19:49Z"
        reason: Completed
        startedAt: "2020-08-18T03:19:49Z"
    name: busybox
    ready: false
    restartCount: 4
    started: false
    state:
      terminated:
        containerID: docker://73e60d2fdb1b322efbd1fdada2d02b7b5e8f2a8b90a9e1a0dfd51516e772d637
        exitCode: 0
        finishedAt: "2020-08-18T03:20:30Z"
        reason: Completed
        startedAt: "2020-08-18T03:20:30Z"
  hostIP: 10.240.0.5
  phase: Running
  podIP: 10.244.1.7
  podIPs:
  - ip: 10.244.1.7
  qosClass: BestEffort
  startTime: "2020-08-18T03:19:04Z"

**Comment by floreks** [comment]
According to the official [documentation](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/).

> The following information is available through environment variables:
>
> status.podIP - the pod’s IP address
> spec.serviceAccountName - the pod’s service account name, available since v1.4.0-alpha.3
> spec.nodeName - the node’s name, available since v1.4.0-alpha.3
> status.hostIP - the node's IP, available since v1.7.0-alpha.1

It will be available as an env var inside the pod, but it will not be possible to get this value at a higher level. Those values are filled out after the container is started. Container description is something like a template that only has access to predefined values, not the ones that are filled out after container is started.

/close
